meson.build

```diff
@@ -91,6 +91,9 @@ if not headers_only
 	mlibc_c_args = ['-Wno-unused-function', '-D__MLIBC_BUILDING_MLIBC', '-nostdinc', '-fno-builtin', '-ffreestanding']
 	mlibc_link_args = ['-nostdlib']
 	add_project_arguments('-Werror=misleading-indentation', language: ['c', 'cpp'])
+	add_project_arguments('-D_DEFAULT_SOURCE', '-D_GNU_SOURCE', language: ['c', 'cpp'])
```

Comment:
All arguments but `_GNU_SOURCE` are certainly wrong; `_DEFAULT_SOURCE` should be set by `mlibc-config.h`, and the rest is set to sane values by default.

The `_GNU_SOURCE` case is more interesting on BSDs - I assume it is not defined there, even for C++?